### PR TITLE
CAMEL-10512-add managedOperations for connection status check for bot…

### PR DESCRIPTION
**Pull request by onders86**
…h consumer&producer

### CODE REVIEW

File: components/camel-mllp/src/main/java/org/apache/camel/component/mllp/MllpTcpClientProducer.java [hunk 1]
```diff
@@ -276,6 +291,19 @@ void checkConnection() throws IOException {
...
+    }
+
+    @ManagedOperation(description = "Check client connection")
+    public boolean managedCheckConnection() {
+        boolean isValid = true;
+        try {
+            checkConnection();
+        } catch (IOException ioEx) {
+            isValid = false;
+            log.error("JMX check connection: {}", ioEx);
```
Comment (aldettinger) [inline components/camel-mllp/src/main/java/org/apache/camel/component/mllp/MllpTcpClientProducer.java]: Are you sure that you want to log an error there ? At first read, it's like this is just nominal case.

File: components/camel-mllp/src/main/java/org/apache/camel/component/mllp/MllpTcpServerConsumer.java [hunk 2]
```diff
@@ -95,6 +98,33 @@ public MllpTcpServerConsumer(MllpEndpoint endpoint, Processor processor) {
...
+                socketAddress = new InetSocketAddress(endpoint.getPort());
+            } else {
+                socketAddress = new InetSocketAddress(endpoint.getHostname(), endpoint.getPort());
+            }
+            Socket checkSocket = new Socket();
+            checkSocket.connect(socketAddress, 100);
+            checkSocket.close();
+        } catch (Exception e) {
+            isValid = false;
+            log.error("JMX check connection: {}", e);
```
Comment (aldettinger) [inline components/camel-mllp/src/main/java/org/apache/camel/component/mllp/MllpTcpServerConsumer.java]: Same thing here.

File: components/camel-mllp/src/main/java/org/apache/camel/component/mllp/MllpTcpClientProducer.java (hunk 1 above)
Comment (davsclaus) [inline components/camel-mllp/src/main/java/org/apache/camel/component/mllp/MllpTcpClientProducer.java]: Yeah it should likely be either DEBUG or ignored IMHO

Comment (onders86) [inline components/camel-mllp/src/main/java/org/apache/camel/component/mllp/MllpTcpClientProducer.java]: well, the reason i wanted to add logging is that there might be different reasons of IOExceptions(SocketException, StreamWriting/Reading exceptions, Connection reset by peer, etc.. ) and the one who wants to manage the application might want to see the stacktrace. Agreed on that i changed to log level as debug which sounds reasonable IMHO.

**Comment by davsclaus** [comment]
Thanks for the PR it has been merged. Do you mind closing this?

**Comment by onders86** [comment]
thanks for accepting. closing.

**Pull request closed without merging**